copter/source/docs/precision-landing-with-irlock.rst

```diff
@@ -1,23 +1,59 @@
 .. _precision-landing-with-irlock:
 
-=========================================
-Precision Landing and Loiter with IR-LOCK
-=========================================
+============================
+Precision Landing and Loiter
+============================
 
 Overview
 ========
 
-Copter supports Precision Landing using the `IR-LOCK sensor <https://irlock.com/collections/frontpage/products/ir-lock-sensor-precision-landing-kit>`__ and a :ref:`sonar or lidar <common-rangefinder-landingpage>`.
-Using this system, when the vehicle enters LAND mode (and has GPS lock) it is possible to reliably land within 30cm of an IR beacon that is moving at less than 1m/s.
+Copter supports Precision Landing using the `IR-LOCK sensor <https://irlock.com/collections/frontpage/products/ir-lock-sensor-precision-landing-kit>`__ and a :ref:`sonar or lidar <common-rangefinder-landingpage>`, or via `MAVLink LANDING_TARGET messages <https://mavlink.io/en/messages/common.html#LANDING_TARGET>`__ from a companion computer providing position differences from a target such as an `April tag <https://docs.wpilib.org/en/stable/docs/software/vision-processing/apriltag/apriltag-intro.html>`__.
 
-Copter also supports Precision Loiter which allows a vehicle to maintain its position above a target while in Loiter mode.  The Pilot can enable this using one of the transmitter's :ref:`auxiliary function switches <common-auxiliary-functions>` (in versions prior to Copter-4.0, CH7_OPT or CH_8_OPT would be used).
+Using this system, when the vehicle enters LAND mode (and has GPS lock) it is possible to reliably land within 30cm of an IR beacon that is moving at less than 1m/s (unless :ref:`PLND_OPTIONS<PLND_OPTIONS>` bit 0 is set to allow a moving landing target) .
+
+Copter also supports Precision Loiter which allows a vehicle to maintain its position above a target while in Loiter mode.  The Pilot can enable this using one of the transmitter's :ref:`auxiliary function switches<common-auxiliary-functions>`.
 
 ..  youtube:: rGFO73ZxADY
     :width: 100%
 
-Where to get it
+
+Precision Loiter demonstration:
+
+..  youtube:: KoLZpSZDfII
+    :width: 100%
+
+Operation
+=========
+Precision Landing, when enabled and active, will provide final positioning during the landing rather than via GPS data only.
+If the landing target is lost (ie position updates with respect to it do not occur), if :ref:`PLND_STRICT<PLND_STRICT>` has been set appropriately, it will try and climb up to a location decided by :ref:`PLND_RET_BEHAVE<PLND_RET_BEHAVE>`. It will then try and come down again in hopes of seeing the landing target. It will keep doing this until :ref:`PLND_RET_MAX<PLND_RET_MAX>` retries have occurred. Then, depending on :ref:`PLND_STRICT<PLND_STRICT>`, it will either just land, or it will stay in the air hovering (useful for, say, landing on boats where you don’t really want to land on water).
+
+Repositioning manually by the pilot during the landing will abort the landing.
```

Comment:
Repositing manually by the pilot will abort precision landing unless PLND_OPTIONS has been set (After Copter 4.3)